### PR TITLE
Adds a new medical machine, the Medical Kiosk.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

So, in an attempt to learn TGUi, I wound up fucking about with the code for long enough that I came up with this as a machine with a relatively simple interface, mostly to show off syntax and basic functionality.

Anyway, this machine serves as a standalone version of the medical analyzer, but can be built anywhere on station to get a basic medical readout, showing brute/burn/toxins/suffocation damage, as well as medical status (Almost always going to say "You're alive."), and if you have some kind of detectable illness. 
MedicalKiosk

Remaining Work:

- [x] Working UI/functionality
- [x] Unique Sprites
- [x] Make Circuit board build-able from the medical protolathe.

## Why It's Good For The Game

Thematically, it fits Medbay to have some way to charge crew for their pre-existing conditions, and otherwise it's beneficial for crew who can't normally get access to a medical analyzer without breaking and entering/powergaming station spawns.

## Changelog
:cl:
add: Medical now has access to a new machine, the Medical Kiosk. For a nominal fee, you can checkup on your basic medical status, without all that risky breaking and entering.
/:cl:

### CODE REVIEW

**Comment by SkeletalElite** [comment]
Where does the money pay to? Medical Budget?

Also maybe in hte future this machine could serve as a better health analyzer to avoid health analyzer bloat. There's already a fucking massive print out in the chat for using the health analyzer as it is. So the handheld could have more basic information for treating minor stuff but this is able to tell you more info at the cost of portability, could be cool.

**Comment by Shapsy** [comment]
I really really like the idea of medical scanners that charge you on a gameplay and fluff level, and I hope this gets expanded on

**Comment by PKPenguin321** [comment]
If you wanna learn the map merger too, I'd like it if these were in the medbay lobby on maps by default

File: code/game/machinery/medical_kiosk.dm [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/obj/machinery/medical_kiosk
```
Comment (PKPenguin321) [inline code/game/machinery/medical_kiosk.dm]: This file is unchecked in the .dme, which is making Travis fail

**Comment by Teepeepee** [comment]
I like this feature and hope it get's implemented, but I have one doubt
do these results get printed on a paper or saved in some way so you can close the UI and still access them, or will I need to exercise my short term memory?

**Comment by ArcaneMusic** [comment]
> Where does the money pay to? Medical Budget?
Currently nowhere, but I'll get it paying to medical.


> If you wanna learn the map merger too, I'd like it if these were in the medbay lobby on maps by default

I'd love to, but for now I'm taking it one step at a time. I only barely figured out TGUi like, a day ago.

File: code/game/machinery/medical_kiosk.dm (hunk 1 above)
Comment (ArcaneMusic) [reply code/game/machinery/medical_kiosk.dm]: Just updated the .dme. On an unrelated note, this is my first PR to include it's own file that would be CHECKED by the .dme. neat.

File: code/game/machinery/medical_kiosk.dm [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  data["brute_health"] = brute_loss
+  data["burn_health"] = fire_loss
+  data["toxin_health"] = tox_loss
+  data["suffocation_health"] = oxy_loss
+  data["patient_status"] = patient_status
+  data["patient_illness"] = sickness
+  data["active_status"] = scan_active ? 0 : 1
+
+  if(user.stat == DEAD || HAS_TRAIT(user, TRAIT_FAKEDEATH))
+    patient_status = "Dead."
```
Comment (YPOQ) [inline code/game/machinery/medical_kiosk.dm]: This needs to be moved up a few lines. It isn't used at this point in the proc.

Comment (ArcaneMusic) [reply code/game/machinery/medical_kiosk.dm]: Whoops. Fixed.

File: code/game/machinery/medical_kiosk.dm [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	var/scan_active = null  //Shows if the machine is being used. resets upon new newer.
+	var/datum/bank_account/account  //payer's account.
+	var/mob/living/carbon/human/H
+	var/obj/item/card/id/C
+	circuit = /obj/item/circuitboard/machine/medical_kiosk
+	payment_department = ACCOUNT_MED
+	var/default_price = 5  //I'm defaulting to a low price on this, but in the future I wouldn't have an issue making it more or less expensive.
+
+/obj/machinery/medical_kiosk/proc/inUse()  //Verifies that the user can use the interface, followed by showing medical information.
+  if(C && C.registered_account)
```
Comment (qwertyquerty) [inline code/game/machinery/medical_kiosk.dm]: This can be `C?.registered_account`

Comment (qwertyquerty) [inline code/game/machinery/medical_kiosk.dm]: This can be `if(C?.registered_account)`

**Comment by Fikou** [comment]
should also tell cloning and brain damage

**Comment by dr-bee** [comment]
> 
> 
> should also tell cloning and brain damage

Those features should be based on the parts used. Tier 4 parts should act like an advanced scanner.

**Comment by carlarctg** [comment]
maybe not t4, just t3 or even 2 considering the price cost and the fact that t4 being made and distributed often takes upwards of 30 minutes

**Comment by Fikou** [comment]
> > should also tell cloning and brain damage
> 
> Those features should be based on the parts used. Tier 4 parts should act like an advanced scanner.

a tier one sleeper/cryo/surgery console tells you cloning damage too i dont remember which one it is i think surgery console but this is an entire machine dedicated to telling your health, it should tell all the damage if you pay for it too

**Comment by ArcaneMusic** [comment]
I'm working on adding brain damage and clone damage working when it's using T3 parts, I'm just running into an issue right now with the adv_scan_active variable I added.

File: code/game/machinery/medical_kiosk.dm [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	..()
+	default_unfasten_wrench(user, I, time = 10)
+	return TRUE
+
+/obj/machinery/medicak_kiosk/RefreshParts()
+  var/A
+  for(var/obj/item/stock_parts/scanning_module/S in component_parts)
+    A += S.rating
+  if(A >=3)
+    adv_scan_active = 1
```
Comment (ArcaneMusic) [inline code/game/machinery/medical_kiosk.dm]: Byond's throwing that adv_scan_active isn't defined, even through I believe I defined it on line 14. Anyone know what's wrong here?

File: code/game/machinery/medical_kiosk.dm [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+		icon_state = "kiosk_off"
+	else
+		icon_state = "kiosk"
+
+/obj/machinery/medical_kiosk/wrench_act(mob/living/user, obj/item/I) //Allows for wrenching/unwrenching the machine.
+	..()
+	default_unfasten_wrench(user, I, time = 10)
+	return TRUE
+
+/obj/machinery/medicak_kiosk/RefreshParts()
```
Comment (optimumtact) [inline code/game/machinery/medical_kiosk.dm]: medical cak kiosk

Comment (ArcaneMusic) [reply code/game/machinery/medical_kiosk.dm]: Blame the Qwerty man for putting K and L so close together

**Comment by AdamElTablawy** [comment]
given that you need to pay to use it it should be better than a handheld scanner

makes sense that an actual machine is better than basically the health iPad

**Comment by ArcaneMusic** [comment]
Alright, I think this is ready. Standard machine interact and deconstruction is working, advanced functions work as well, and it's now on the correct item layer. I'll touch adding it to the maps later when I don't have a week full of exams to work on this upcoming week, but it's build-able from the medical lathe in the meantime after basic medical tech is reached.

**Comment by spessbandit** [comment]
>introducing capitalism into medbay
fuck this, space medicare for all

**Comment by Fhaxaris** [comment]
people still gonna greytide the medical room

**Comment by Jalleo** [comment]
@ArcaneMusic
For reference with mapping you start the mapmerger.bat select the map you want to map(s) merge. Hit enter. Then open it up in dream maker and make the change. 
 (shift left click if tile has other stuff by default I believe to not remove other items)
Save the map.
Hit enter on mapmerger again and it should be ready to use.
You do require python to be installed. Its mpre annoying if you forget the first step. 
(If you do just copy the map post save elsewhere undo changes then start the map merge process instead.)

File: code/game/machinery/medical_kiosk.dm [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+/obj/machinery/medical_kiosk/attackby(obj/item/O, mob/user, params)
+  if(default_deconstruction_screwdriver(user, "kiosk_open", "kiosk", O))
+    return
+
+  if(default_pry_open(O))
+    return
+
+  if(default_deconstruction_crowbar(O))
+    return
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/medical_kiosk.dm]: a-a-are you just wanting it to return by default?

Comment (ArcaneMusic) [reply code/game/machinery/medical_kiosk.dm]: I was basing this off other machine code, didn't realize every other one defaults a return. Is this what you meant?

**Pull request merged**